### PR TITLE
Cpa 67 fit poly vector3

### PR DESCRIPTION
**Pull request by EMullerMDA**

### CODE REVIEW

File: modules/c++/math.poly/include/math/poly/Fit.h [hunk 1]
```diff
@@ -250,6 +251,79 @@ inline math::poly::TwoD<double> fit(size_t numRows,
 
     return fit(xm, ym, zm, nx, ny);
 }
+
+typedef math::linear::VectorN<3, double> Vector3;
+
+inline math::poly::OneD<Vector3> fit(
```
Comment (asylvest) [inline modules/c++/math.poly/include/math/poly/Fit.h]: For this and all the functions in here, use Doxygen style comments along these lines:
```
    /*!
     * Returns a scaled polynomial such that
     * P'(x, y) = P(x * scaleX, y * scaleY)
     *
     * \param scaleX The factor to apply to x
     * \param scaleY The factor to apply to y
     *
     * \return Scaled polynomial of the same order as the original polynomial
     */
```

File: modules/c++/math.poly/include/math/poly/Fit.h [hunk 2]
```diff
@@ -250,6 +251,79 @@ inline math::poly::TwoD<double> fit(size_t numRows,
...
+    if (yObs0.size() != numObs || yObs1.size() != numObs || yObs2.size() != numObs)
+    {
+        throw except::Exception(Ctxt("Must have the same number of observed y values as observed x values"));
+    }
+    if (numCoeffs < 1)
+    {
+        throw except::Exception(Ctxt("Number of coefficients must be >= 1"));
+    }
+
+    math::poly::OneD<double> fit0 = fit(xObs, yObs0, numCoeffs);
```
Comment (asylvest) [inline modules/c++/math.poly/include/math/poly/Fit.h]: fit0, fit1, and fit2 can all be const

File: modules/c++/math.poly/include/math/poly/Fit.h [hunk 3]
```diff
@@ -250,6 +251,79 @@ inline math::poly::TwoD<double> fit(size_t numRows,
...
+    math::poly::OneD<double> fit1 = fit(xObs, yObs1, numCoeffs);
+    math::poly::OneD<double> fit2 = fit(xObs, yObs2, numCoeffs);
+
+    math::poly::OneD<Vector3> polyVector3 = math::poly::OneD<Vector3>(numCoeffs-1);
+    for (size_t i = 0; i< numCoeffs; i++)
+    {
+        Vector3 coeffs = Vector3();
```
Comment (asylvest) [inline modules/c++/math.poly/include/math/poly/Fit.h]: Change `i` to `ii` (avoid single character variable names)
Also, can't you say `Vector3& coeffs = polyVector3[i]` and then avoid the copy and last line?

File: modules/c++/math.poly/include/math/poly/Fit.h [hunk 4]
```diff
@@ -250,6 +251,79 @@ inline math::poly::TwoD<double> fit(size_t numRows,
...
+    return polyVector3;
+}
+
+inline math::poly::OneD<Vector3> fit(
+    const math::linear::Vector<double>& xObsVector,
+    const math::linear::Matrix2D<double>& yObsMatrix,
+    size_t numCoeffs)
+{
+    // Vector size error checking will be done by the base fit() function
+    size_t numObs = yObsMatrix.cols();
```
Comment (asylvest) [inline modules/c++/math.poly/include/math/poly/Fit.h]: Can be const

**Comment by asylvest** [comment]
Looks good - just all trivial stuff

**Comment by EMullerMDA** [comment]
Code review changes made.

**Pull request merged**